### ISSUE TITLE
[Prio: Low] Removal of cnpg pods sometimes lead to CNPG manager getting stuck

### ISSUE THREAD

**Issue by Ornias1993** [cloudnative-pg/cloudnative-pg#1947]
In most cases (accidental) removal of a CNPG pods, leads to the manager restoring the pod right away (as it should).

However, in some edge cases, it gets into a limbo state not restoring them:
```
{"level":"info","ts":"2023-04-23T09:59:35Z","msg":"No primary instance found for this cluster","controller":"cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster","Cluster":{"name":"vaultwarden2-cnpg-main","namespace":"ix-vaultwarden2"},"namespace":"ix-vaultwarden2","name":"vaultwarden2-cnpg-main","reconcileID":"348b5fdd-fbbf-4925-afa6-ae9049c8517b","uuid":"8d3fe8e2-e1bd-11ed-b82c-aa1881d0aa5a"}
```

```
vaultwarden2-cnpg-main-rw-b969898d5-ktx8l   1/1     Running   0             9m17s
vaultwarden2-cnpg-main-3-join-2nf8f         1/1     Running   0             9m15s
vaultwarden2-cnpg-main-rw-b969898d5-44cjh   1/1     Running   0             2m4s
vaultwarden2-88b49f588-vwfrb                0/1     Running   3 (43s ago)   2m3s
```

**Comment by Ornias1993** [comment]
We, as @truecharts, have found a solution to fix clusters in this limbo state:

- Hibernate the cluster
- Wait 5 minutes
- Unhibernate the cluster


This forces the controller to completely go over the cluster again as-if it was freshly starting, including ensuring there is a primary instance.

**Comment by Ornias1993** [comment]
Adding prio: low tot the title, as this is basically an issue that only occurs when users make mistakes and a quickfix/workaround has been found.

Leaving the issue as this should idealy be a scenario the operator detect itself

**Comment by jonaskello** [comment]
I encountered a similar situation where I deleted one of the pods and the cluster got totally stuck and I could not find any way out of it. I tried hibernating but it did not work. I think #1915 is also about this

**Comment by Ornias1993** [comment]
Ensure you hibernate AND scale down the pooler though...

**Comment by Nepoxx** [comment]
> We, as @truecharts, have found a solution to fix clusters in this limbo state:
> 
> * Hibernate the cluster
> * Wait 5 minutes
> * Unhibernate the cluster
> 
> This forces the controller to completely go over the cluster again as-if it was freshly starting, including ensuring there is a primary instance.

How do you hibernate a cluster without a primary?

I'm stuck in the same state and trying your workaround gives me this `Error: no primary instance found, cannot proceed with the hibernation`

**Comment by sxd** [comment]
This should be fixed in the latest version 1.22.2, please open a new issue if it happens again
